src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CodeActionEndToEndTest.cs

```diff
@@ -341,19 +361,22 @@ public async Task Handle_ConvertToInterpolatedString_CodeBlock()
     #region RazorCodeAction Tests
 
     [Theory]
-    [InlineData("[||]DoesNotExist")]
-    [InlineData("Does[||]NotExist")]
-    [InlineData("DoesNotExist[||]")]
-    public async Task Handle_GenerateMethod_NoCodeBlock(string cursorAndMethodName)
+    [InlineData("[||]DoesNotExist", GenerateEventHandlerReturnType, GenerateEventHandlerTitle)]
+    [InlineData("Does[||]NotExist", GenerateEventHandlerReturnType, GenerateEventHandlerTitle)]
+    [InlineData("DoesNotExist[||]", GenerateEventHandlerReturnType, GenerateEventHandlerTitle)]
+    [InlineData("[||]DoesNotExist", GenerateAsyncEventHandlerReturnType, GenerateAsyncEventHandlerTitle)]
+    [InlineData("Does[||]NotExist", GenerateAsyncEventHandlerReturnType, GenerateAsyncEventHandlerTitle)]
+    [InlineData("DoesNotExist[||]", GenerateAsyncEventHandlerReturnType, GenerateAsyncEventHandlerTitle)]
+    public async Task Handle_GenerateMethod_NoCodeBlock(string cursorAndMethodName, string returnType, string codeActionTitle)
```

Comment:
`<opinion>` 

I love the use of inline data here, but it's getting a bit harder to follow all of the combinations. I suggest either of the following: 

```csharp
[Theory]
[InlineData("[||]DoesNotExist")]
// rest here
public async Task Handle_GenerateMethod_NoCodeBlock(...)
{
}

[Theory]
[InlineData("[||]DoesNotExist")]
// rest here
public async Task Handle_GenerateAsyncMethod_NoCodeBlock(...)
{
}
```

or...

```csharp
static readonly (string, string) VoidEventHandlerInfo = (GenerateEventHandlerReturnType, GenerateEventHandlerTitle);
static readonly (string, string) AsyncEventHandlerInfo = (GenerateAsyncEventHandlerReturnType, GenerateAsyncEventHandlerTitle);

[Theory]
[InlineData("[||]DoesNotExist", VoidEventHandlerInfo)]
[InlineData("[||]DoesNotExist", AsyncEventHandlerInfo)]
// rest here
public async Task Handle_GenerateMethod_NoCodeBlock(string cursorAndMethodName, (string returnType, string title) methodInfo)
{
}
```

_please forgive my browser code_ 

It ends up pairing better and is a little easier to add more options later 

`</opinion>`

Reply:
Good point! I decided to just separate them per your first suggestion.